patches/0109-lavf-dnn-enable-libtorch-backend-support-FRVSR-model.patch

```diff
@@ -0,0 +1,141 @@
+From 94c391506bfd8ebd01b8843724fa055e7995a4e7 Mon Sep 17 00:00:00 2001
+From: Ting Fu <ting.fu@intel.com>
+Date: Wed, 29 Jun 2022 07:46:44 +0000
+Subject: [PATCH 4/4] lavf/dnn: enable libtorch backend support FRVSR model
+
+Signed-off-by: Ting Fu <ting.fu@intel.com>
+---
+ libavfilter/dnn/dnn_backend_torch.cpp | 53 ++++++++++++++++++++++++---
+ 1 file changed, 47 insertions(+), 6 deletions(-)
+
+diff --git a/libavfilter/dnn/dnn_backend_torch.cpp b/libavfilter/dnn/dnn_backend_torch.cpp
+index c5592a4b4e..043fdfb4ca 100644
+--- a/libavfilter/dnn/dnn_backend_torch.cpp
++++ b/libavfilter/dnn/dnn_backend_torch.cpp
+@@ -46,6 +46,8 @@ typedef struct THContext {
+     THOptions options;
+ } THContext;
+ 
++typedef enum {UNSUPPORTED_MODEL = -1, BASICVSR, FRVSR} ModelType;
++
+ typedef struct THModel {
+     THContext ctx;
+     DNNModel *model;
+@@ -53,6 +55,7 @@ typedef struct THModel {
+     SafeQueue *request_queue;
+     Queue *task_queue;
+     Queue *lltask_queue;
++    ModelType model_type;
+ } THModel;
+ 
+ typedef struct THInferRequest {
+@@ -213,6 +216,7 @@ DNNModel *ff_dnn_load_model_th(const char *model_filename, DNNFunctionType func_
+ 
+     th_model->ctx.c_class = &dnn_th_class;
+     ctx = &th_model->ctx;
++    th_model->model_type = UNSUPPORTED_MODEL;
+     //parse options
+     av_opt_set_defaults(ctx);
+     if (av_opt_set_from_string(ctx, options, NULL, "=", "&") < 0) {
+@@ -242,6 +246,7 @@ DNNModel *ff_dnn_load_model_th(const char *model_filename, DNNFunctionType func_
+         av_log(ctx, AV_LOG_ERROR, "Failed to load torch model\n");
+         goto fail;
+     }
++    const torch::jit::NameTensor& first_param = *th_model->jit_model.named_parameters().begin();
+ 
+     th_model->request_queue = ff_safe_queue_create();
+     if (!th_model->request_queue) {
+@@ -276,6 +281,15 @@ DNNModel *ff_dnn_load_model_th(const char *model_filename, DNNFunctionType func_
+         goto fail;
+     }
+ 
++    if (!first_param.name.find("fnet")) {
++        th_model->model_type = FRVSR;
++    } else if (!first_param.name.find("spynet")) {
++        th_model->model_type = BASICVSR;
++    } else {
++        th_model->model_type = UNSUPPORTED_MODEL;
++                avpriv_report_missing_feature(NULL, "Unsupported model type\n");
++        goto fail;
++    }
+     th_model->model = model;
+     model->model = th_model;
+     model->get_input = &get_input_th;
+@@ -341,8 +355,13 @@ static int fill_model_input_th(THModel *th_model, THRequestItem *request)
+         avpriv_report_missing_feature(NULL, "model function type %d", th_model->model->func_type);
+         break;
+     }
+-    *infer_request->input_tensor = torch::from_blob(input.data, {1, 1, 3, input.height, input.width},
+-                                                    torch::kFloat32);
++    if (th_model->model_type == FRVSR) {
```

Comment:
You can add a other type, otherwise pytorch plugins can only support BasicVSR and FRVSR.

Reply:
Sure, i changed it into if else clause, no BASICVSR only check.